astroquery/mast/observations.py

```diff
@@ -532,10 +532,12 @@ def download_file(self, uri, *, local_path=None, base_url=None, cache=True, clou
         base_url = base_url if base_url else self._portal_api_connection.MAST_DOWNLOAD_URL
         data_url = base_url + "?uri=" + uri
 
-        # create a local file path if none is input.  Use current directory as default.
-        if not local_path:
-            filename = os.path.basename(uri)
-            local_path = os.path.join(os.path.abspath('.'), filename)
+        # parse a local file path from local_path parameter.  Use current directory as default.
+        filename = os.path.basename(uri)
+        if not local_path:  # local file path is not defined
+            local_path = filename
+        elif os.path.isdir(local_path):  # local file path is directory
```

Comment:
It also might need a second `elif` to finish this off: in the case that `local_path` is supplied as a filename?

Reply:
In the most recent changes, I differentiate between a filename and a directory by checking if the `pathlib.Path` object has an extension. I don't think we should need a branch to check that `local_path` is a filename since the parameter shouldn't need any further processing and can be passed directly into the `download_file` function.